plugins/modules/purefa_network.py

```diff
@@ -268,53 +378,117 @@ def update_interface(module, array, interface):
         full_addr = address + "/" + cidr
         if module.params["gateway"] not in IPNetwork(full_addr):
             module.fail_json(msg="Gateway and subnet are not compatible.")
         gateway = module.params["gateway"]
-    if ip_version == "6":
-        netmask = str(IPAddress(netmask).netmask_bits())
+    else:
+        gateway = module.params["gateway"]
+
     new_state = {
+        "enabled": enabled,
         "address": address,
         "mtu": mtu,
         "gateway": gateway,
         "netmask": netmask,
         "services": services,
+        "slaves": subinterfaces,
     }
+    if (
+        current_state["address"]
+        and IPAddress(new_state["address"]).version
+        != IPAddress(current_state["address"]).version
+    ):
+        if (
+            IPAddress(new_state["gateway"]).version
+            != IPAddress(new_state["address"]).version
+        ):
+            module.fail_json(
+                msg="Changing IP protocol requires gateway to change as well."
+            )
     if new_state != current_state:
         changed = True
+        if (
+            module.params["servicelist"]
+            and sorted(module.params["servicelist"]) != interface["services"]
+        ):
+            api_version = array._list_available_rest_versions()
+            if FC_ENABLE_API in api_version:
+                if HAS_PYPURECLIENT:
+                    if not module.check_mode:
+                        network = NetworkInterfacePatch(
+                            services=module.params["servicelist"]
+                        )
+                        res = array6.patch_network_interfaces(
+                            names=[module.params["name"]], network=network
+                        )
+                        if res.status_code != 200:
+                            module.fail_json(
+                                msg="Failed to update interface service list {0}. Error: {1}".format(
+                                    module.params["name"], res.errors[0].message
+                                )
+                            )
+                else:
+                    module.warn_json(
+                        "Servicelist not updated as pypureclient module is required"
+                    )
         if (
             "management" in interface["services"] or "app" in interface["services"]
-        ) and address == "0.0.0.0/0":
+        ) and address in ["0.0.0.0/0", "::/0"]:
             module.fail_json(
                 msg="Removing IP address from a management or app port is not supported"
             )
         if not module.check_mode:
             try:
+                array.set_network_interface(
+                    interface["name"], enabled=new_state["enabled"]
+                )
                 if new_state["gateway"] is not None:
                     array.set_network_interface(
                         interface["name"],
+                        enabled=new_state["enabled"],
```

Comment:
This causes the `try` block to file. I think you had discovered this which is why the `set_network_interface` call is made just above to enable the interface.

I think to fix this, you just need to remove the `enabled=new_state["enabled"],` line.

Reply:
Fixed